File: tdd-workflow-example-php/test/Greeting/GreeterTest.php
<?php

namespace Greeting;

class GreeterTest extends \PHPUnit_Framework_TestCase
{
    /** @var Greeter */
    private $sut;

    protected function setUp()
    {
        $this->sut = new Greeter();
    }

    /** @dataProvider provideGreetingData */
    public function testGreet(string $expected, string $given)
    {
        static::assertEquals($expected, $this->sut->greet($given));
    }

    public function provideGreetingData()
    {
        return [
            ['expected' => 'Hello, World!', 'given' => 'World'],
            ['expected' => 'Hello, Tests!', 'given' => 'Tests'],
            ['expected' => 'Hi!', 'given' => ''],
        ];
    }

    /** @dataProvider provideLoudGreetingData */
    public function testLoudGreet(string $expected, string $given)
    {
        static::assertEquals($expected, $this->sut->loudGreet($given));
    }

    public function provideLoudGreetingData()
    {
        return [
            ['expected' => 'HELLO, WORLD!', 'given' => 'World'],
            ['expected' => 'HELLO, TESTS!', 'given' => 'Tests'],
            ['expected' => 'HI!', 'given' => ''],
        ];
    }
}

File: tdd-workflow-example-java/src/test/java/greeting/GreeterTest.java
package greeting;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;

import static org.junit.Assert.*;

@RunWith(Parameterized.class)
public class GreeterTest {
    private Greeter sut;

    public String expected;
    public String given;

    @Parameterized.Parameters(name = "name={1}: greet({1})={0}")
    public static Iterable<Object[]> data() {
        return Arrays.asList(new Object[][]{
                {"Hello, World!", "World"},
                {"Hello, Tests!", "Tests"},
                {"Hi!", ""}
        });
    }

    @Before
    public void setUp() {
        sut = new Greeter();
    }

    public GreeterTest(String expected, String given) {
        this.expected = expected;
        this.given = given;
    }

    @Test
    public void greet() {
        assertEquals(expected, sut.greet(given));
    }

    @Test
    public void loudGreet() {
        assertEquals(expected.toUpperCase(), sut.loudGreet(given));
    }
}
